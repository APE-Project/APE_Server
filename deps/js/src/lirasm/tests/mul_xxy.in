; 1073741823 * 2 = 0x7ffffffe, and will nearly (but not quite) overflow.
big = immi 1073741823
two = immi 2

; Because 'big' isn't used after mul, it _may_ get allocated to the same
; register as 'res'. This is the case with the ARM back-end, and that is where
; this test is important as rX=rX*rY isn't possible on ARMv5 without some
; trickery.
res = mulxovi big two    ; no overflow, so we don't exit here

; Store 'res' so it isn't dead.
m = allocp 4
sti res m 0
x                       ; we exit here
