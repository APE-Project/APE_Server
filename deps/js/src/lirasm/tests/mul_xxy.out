Exited block on line: 14
